src/components/App.js

```diff
@@ -16,25 +19,100 @@ import '../style.css';
 		*Provide form to input repo information dynamically, and display related issues
 */
 
+const options = [
+    {
+      label: "All",
+      value: "all",
+      size: "80px"
+    },
+    {
+      label: "Open",
+      value: "open",
+      size: "100px"
+    },
+    {
+      label: "Closed",
+      value: "closed",
+      size: "110px"
+    },
+  ];
+
 const App = () => {
     
     const [issue, setIssue] = useState([]);
+    const [label, setLabel] = useState([]);
+    const [pageNumber, setpageNumber] = useState(0);
+    const [githubState, setgithubState] = useState(options[0].value); // for filter open/closed/all
+
+    const loadIssues = async (pageNumber) => {
+        let gitState = '';
+            if(githubState === 'all') {
+                gitState = '&state=all'
+            } else {
+                gitState = '&state=' + githubState.value
+            }
 
-        const loadIssues = async () => {
-            const response = await github.get();
+        if (pageNumber === 0) {
+            setpageNumber(1)
+            const response = await github.get('/issues?page=' + pageNumber +gitState);
+            setIssue(response.data)
+        } else {
+            const response = await github.get('/issues?page=' + pageNumber +gitState);
+            if(response.data.length  === 0){
+                setpageNumber(pageNumber -1)
+                const response = await github.get('/issues?page=' + pageNumber +gitState);
+                setIssue(response.data)
+            }
             setIssue(response.data)
         }
```

Comment:
```suggestion
    const response = await github.get('/issues', {
      params: {
        page:  pageNumber ?? 1,
        state: githubState === 'all' ? 'all' : githubState.value
      }
    });
```
All these lines can be simplified by this

Reply:
thank you kaayo sir. learned so much here.